pkg/core/resources/apis/mesh/proxytemplate_validator.go

```diff
@@ -0,0 +1,80 @@
+package mesh
+
+import (
+	"fmt"
+	"github.com/Kong/kuma/api/mesh/v1alpha1"
+	"github.com/Kong/kuma/pkg/core/validators"
+	"github.com/Kong/kuma/pkg/xds/template"
+	"strings"
+)
+
+var availableProfiles map[string]bool
+var availableProfilesMsg string
+
+func init() {
+	profiles := []string{}
+	availableProfiles = map[string]bool{}
+	for _, profile := range template.AvailableProfiles {
+		availableProfiles[profile] = true
+		profiles = append(profiles, profile)
+	}
+	availableProfilesMsg = strings.Join(profiles, ",")
+}
+
+func (t *ProxyTemplateResource) Validate() error {
+	var verr validators.ValidationError
+	verr.AddError("", validateImports(t.Spec.Imports))
+	verr.AddError("", validateResources(t.Spec.Resources))
+	verr.AddError("", validateSelectors(t.Spec.Selectors))
+	return verr.OrNil()
+}
+
+func validateImports(imports []string) validators.ValidationError {
+	var verr validators.ValidationError
+	for i, imp := range imports {
+		if imp == "" {
+			verr.AddViolationAt(validators.RootedAt("imports").Index(i), "cannot be empty")
+			continue
+		}
+		if !availableProfiles[imp] {
+			verr.AddViolationAt(validators.RootedAt("imports").Index(i), fmt.Sprintf("profile not found. Available profiles: %s", availableProfilesMsg))
+		}
+	}
+	return verr
+}
+
+func validateResources(resources []*v1alpha1.ProxyTemplateRawResource) validators.ValidationError {
+	var verr validators.ValidationError
+	for i, resource := range resources {
+		if resource.Name == "" {
+			verr.AddViolationAt(validators.RootedAt("resources").Index(i).Field("name"), "cannot be empty")
+		}
+		if resource.Version == "" {
+			verr.AddViolationAt(validators.RootedAt("resources").Index(i).Field("version"), "cannot be empty")
+		}
+		if resource.Resource == "" {
+			verr.AddViolationAt(validators.RootedAt("resources").Index(i).Field("resource"), "cannot be empty")
+		} else if _, err := template.ResourceFromYaml(resource.Resource); err != nil {
+			verr.AddViolationAt(validators.RootedAt("resources").Index(i).Field("resource"), err.Error())
+		}
+	}
+	return verr
+}
+
+func validateSelectors(selectors []*v1alpha1.ProxyTemplate_Selector) validators.ValidationError {
```

Comment:
Let's reuse `validators.ValidateSelectors()`.

For that, we will need to:
* add `RequireServiceTag` flag to `ValidateSelectorOpts`
* add a separate `Opts` struct for `validators.ValidateSelectors()`:
  ```
  type ValidateSelectorsOpts struct {
    ValidateSelectorOpts
    RequireAtLeastOneSelector bool
  } 
  ```